### PR TITLE
Standardize usage of unique names throughout tests

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?

* Adds a utility function for getting unique names for test entities.
* Adds Azure build Id to created entities names to be able to match potential leftover entities to Azure builds.

### Review checklist
Please check relevant items below:
- [x] This PR includes all [newly recorded cassettes](/DEVELOPMENT.md) for any modified tests.

- [x] This PR does not rely on API client schema changes.
    - [ ] The CI should be fully passing.
- [ ] Or, this PR relies on API schema changes and this is a Draft PR to include tests for that new functionality.
  - Note: CI shouldn't be run on this Draft PR, as its expected to fail without the corresponding schema changes.

### CODE REVIEW

File: tests/test_utils.go [hunk 1]
```diff
@@ -146,6 +147,23 @@ func WithClock(ctx context.Context, t *testing.T) context.Context {
 	return context.WithValue(ctx, clockKey, fc)
 }
 
+// UniqueEntityName will return a unique string that can be used as a title/description/summary/...
+// of an API entity. When used in Azure Pipelines, it will include BuildId to enable mapping resources
+// that weren't deleted to builds.
+func UniqueEntityName(ctx context.Context, t *testing.T) *string {
+	buildID, present := os.LookupEnv("BUILD_BUILDID")
```
Comment (jirikuncar) [inline tests/test_utils.go:154]: 1. only when `IsRecording` or better `RECORD=none`
2. only when `CI=true`

File: tests/api/v1/datadog/api_aws_integration_test.go [hunk 2]
```diff
@@ -19,10 +18,9 @@ import (
 	"gopkg.in/h2non/gock.v1"
 )
 
-func generateUniqueAWSAccount(ctx context.Context) datadog.AWSAccount {
-	accountID := fmt.Sprintf("go_%09d", tests.ClockFromContext(ctx).Now().UnixNano()%1000000000)
+func generateUniqueAWSAccount(ctx context.Context, t *testing.T) datadog.AWSAccount {
 	return datadog.AWSAccount{
-		AccountId:                     &accountID,
+		AccountId:                     tests.UniqueEntityName(ctx, t),
```
Comment (zippolyte) [inline tests/api/v1/datadog/api_aws_integration_test.go:23]: I think accountID needs to be 12 chars exactly, so this probably won't work.
maybe we can have this in the role name, or as a tag in order to identify from where some account ids come from ?

File: tests/test_utils.go [hunk 3]
```diff
@@ -146,6 +147,23 @@ func WithClock(ctx context.Context, t *testing.T) context.Context {
...
+		buildID = "local"
+	}
+
+	// NOTE: some endpoints have limits on certain fields (e.g. Roles V2 names can only be 55 chars long),
+	// so we need to keep this short
+    result := fmt.Sprintf("go-%s-%s-%d", t.Name(), buildID, ClockFromContext(ctx).Now().Unix())
+    // In case this is used in URL, make sure we replace the slash that is added by subtests
+    result = strings.ReplaceAll(result, "/", "-")
+    return &result
+}
```
[the remark this answers is not in the document]
Comment (bkabrda) [reply tests/test_utils.go:165]: Hmm, good point. My IDE seems to be messing this up in some places and I don't know why :/ Will fix it.

File: tests/api/v1/datadog/api_aws_integration_test.go (hunk 2 above)
Comment (bkabrda) [reply tests/api/v1/datadog/api_aws_integration_test.go:23]: Hmm, but the tests are passing, so it seems it does work like this 🤔

**Comment by bkabrda** [comment]
The Azure tests are failing now on `TestUpdateRoleErrors/404_Bad_Role_ID_in_Path`, which is expected and will be fixed by #292, so I'm merging this.

**Pull request merged**